client/actions/book_actions.js

```diff
@@ -0,0 +1,77 @@
+import axios from 'axios';
+
+import { ADD_BOOK, GET_ALL_BOOKS, GET_RENTED_BOOKS } from './types';
+
+const API_URL = 'http://localhost:8000/api/v1/books',
+  USER_API_URL = 'http://localhost:8000/api/v1/users';
+
+export function addNewBook(bookDetails) {
+  return dispatch => axios.post(`${API_URL}`, bookDetails)
+    .then((res) => {
+      dispatch({
+        type: ADD_BOOK,
+        message: res.data.message
+      });
+    })
+    .catch(error => error);
+}
+
+export function getAllBooks() {
+  return dispatch => axios.get(`${API_URL}`)
+    .then((res) => {
+      dispatch({
+        type: GET_ALL_BOOKS,
+        data: res.data
+      });
+      return res.data;
+    })
+    .catch(error => error);
+}
+
+export function deleteBook(bookId) {
+  return axios.delete(`${API_URL}/delete/${bookId}`)
+    .then(res => res.data.message)
+    .catch(error => error);
+}
```

Comment:
Are you going to send the whole error object to the user? You may want to send just the error message.

Reply:
Yes, you are right, I filtered that out when displaying it to the user.